src/oop/special_class_method/hard/q13/MainTest.java

```diff
@@ -0,0 +1,156 @@
+        assertTrue(testClass.hasField(attributeName2, double.class),
+                "Your " + attributeName2 + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(attributeName2, "private"),
+                "Your " + attributeName2 + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(attributeName3),
+                "Your " + className + " class is missing the " + attributeName3 + " attribute, or it is spelled incorrectly.");
+        assertTrue(testClass.hasField(attributeName3, String.class),
+                "Your " + attributeName3 + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(attributeName3, "private"),
+                "Your " + attributeName3 + " does not have the correct visibility modifier.");
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("Red", 3423.4, "Milk", "Water"),
+                Arguments.of("Black", 54.5, "Water", "Lava"),
+                Arguments.of("Blue", 934.3, "Mercury", "Juice"),
+                Arguments.of("White", 49374.43, "Blood", "Water"),
+                Arguments.of("Yellow", 434.43, "Lava", "Whiskey")
+        );
+    }
+
+
+    @Test
+    public void cupClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {String.class, double.class, String.class};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct visibility modifiers.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void cupClassHasCorrectConstructor(String value1, double value2, String value3) throws Throwable {
+        Object[][] arguments = {
+                {value1, String.class},
+                {value2, double.class},
+                {value3, String.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        _assertEquals(value1, testClass.getFieldValue(classInstance, attributeName1),
+                "Your " + className + " constructor does not initialize the " + attributeName1 + " attribute to the correct value.");
+        _assertEquals(value2, testClass.getFieldValue(classInstance, attributeName2),
+                "Your " + className + " constructor does not initialize the " + attributeName2 + " attribute to the correct value.");
+        _assertEquals(value3, testClass.getFieldValue(classInstance, attributeName3),
+                "Your " + className + " constructor does not initialize the " + attributeName3 + " attribute to the correct value.");
+    }
+
+    @Test
+    public void getVolumeIsDefinedCorrectly() {
```

Comment:
You can just remove this test as you cover it in the `correctGetVolumeMethod` test.